### PR TITLE
Abnormal security

### PR DESCRIPTION
**Pull request by dschuan**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.
Contains an integration that allows users to pull Abnormal Security threats and cases into XSOAR.

## Screenshots
Paste here any images that will help the reviewer
where the integration appears

cli commands example

## Minimum version of Cortex XSOAR
- [ ] 5.5.0
- [x] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Comment by marcellarichmond** [comment]
@mchasepan fyi

**Comment by dschuan** [comment]
Just completed the contribution registration a couple hours earlier.

**Review by Shellyber — COMMENTED** [review]
Looks very good! 👍 
Few more notes:
* Please change the integration name and its files to be AbnormalSecurity
* All the arguments and rhe params should be in the form of `page_size`, `threat_id` and not pageSize, threatId - Please replace everywhere its needed
* Context outputs should start with `AbnormalSecurity`
* When a command is getting an argument of id, please specify in its description where can the user can get it from.
* Don't forget to update the readme file once the changes are done.

**Comment by Shellyber** [comment]
@dschuan  Let's also schedule a demo for the integration.
For what I saw in the video, it is recommended to return the results to the war room using `tableToMarkdown`. 
It will look better to the users. 
Nice and clean table that contains every result in a raw
https://xsoar.pan.dev/docs/integrations/context-and-outputs#return-results-with-custom-markdown

File: Packs/AbnormalSecurity/Integrations/AbxCortexXSOAR/AbxCortexXSOAR.py [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+
+
+def main():
+    params = demisto.params()
+    args = demisto.args()
+    url = params.get('url')
+    verify_certificate = not params.get('insecure', False)
+    proxy = params.get('proxy', False)
+    headers = {}
+    mock_data = str(args.get('mock-data', ''))
```
[the remark this answers is not in the document]
Comment (dschuan) [reply Packs/AbnormalSecurity/Integrations/AbxCortexXSOAR/AbxCortexXSOAR.py:259]: This is a requirement from our customers as they have use cases where they wish to retrieve mock data to generate their own custom test playbooks

File: Packs/AbnormalSecurity/Integrations/AbxCortexXSOAR/AbxCortexXSOAR.yml [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+      name: mock-data
+      required: false
+    description: Check the status of an action requested on a threat.
+    name: abxcortexxsoar-check-the-status-of-an-action-requested-on-a-threat
+    outputs:
+    - contextPath: AbxCortexXSOAR.ActionStatus.status
+      description: ''
+      type: String
+    - contextPath: AbxCortexXSOAR.ActionStatus.description
+      description: ''
```
[the remark this answers is not in the document]
Comment (dschuan) [reply Packs/AbnormalSecurity/Integrations/AbxCortexXSOAR/AbxCortexXSOAR.yml:74]: Our customers requested for this description field to be added in their use cases

[the remark this answers is not in the document]
Comment (Shellyber) [reply Packs/AbnormalSecurity/Integrations/AbxCortexXSOAR/AbxCortexXSOAR.yml:74]: ok no problem. 
so please add here the description.  we do not leave it empty.

**Comment by dschuan** [comment]
Thanks for the review and change recommendations, will take them into account and push out a new iteration after testing out the changes. Sent you a tentative invite via email, please let me know if you're unavailable and we can reschedule if necessary. Cheers!

File: Packs/AbnormalSecurity/Integrations/AbnormalSecurity/AbnormalSecurity.yml [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+    - description: Action to perform on threat
+      isArray: false
+      name: action
+      required: true
+    - description: Returns test data if set to `True`
+      isArray: false
+      name: mock-data
+      required: false
+    description: Manage a Threat identified by Abnormal Security
+    name: abnormal-security-manage-threat
```
Comment (Shellyber) [inline Packs/AbnormalSecurity/Integrations/AbnormalSecurity/AbnormalSecurity.yml:301]: Missing outputs for this command

File: Packs/AbnormalSecurity/Integrations/AbnormalSecurity/AbnormalSecurity.yml [hunk 4]
```diff
@@ -0,0 +1,340 @@
...
+    - description: Action to perform on case
+      isArray: false
+      name: action
+      required: true
+    - description: Returns test data if set to `True`
+      isArray: false
+      name: mock-data
+      required: false
+    description: Manage an Abnormal Case.
+    name: abnormal-security-manage-abnormal-case
```
Comment (Shellyber) [inline Packs/AbnormalSecurity/Integrations/AbnormalSecurity/AbnormalSecurity.yml:316]: Same, missing outputs for this command

File: Packs/AbnormalSecurity/Integrations/AbnormalSecurity/AbnormalSecurity.yml [hunk 5]
```diff
@@ -0,0 +1,340 @@
...
+      isArray: false
+      name: reporter
+      required: true
+    - description: Type of misjudgement reported
+      isArray: false
+      name: report_type
+      required: true
+    description: Submit an Inquiry to request a report on misjudgement by Abnormal
+      Security
+    name: abnormal-security-submit-inquiry-to-request-a-report-on-misjudgement-by-abnormal-security
```
Comment (Shellyber) [inline Packs/AbnormalSecurity/Integrations/AbnormalSecurity/AbnormalSecurity.yml:332]: same as above.

File: Packs/AbnormalSecurity/Integrations/AbnormalSecurity/AbnormalSecurity.py [hunk 6]
```diff
@@ -0,0 +1,331 @@
...
+            'abnormal-security-get-threat':
+                get_details_of_a_threat_command,
+            'abnormal-security-get-abnormal-case':
+                get_details_of_an_abnormal_case_command,
+            'abnormal-security-get-latest-threat-intel-feed': get_the_latest_threat_intel_feed_command,
+            'abnormal-security-manage-threat':
+                manage_a_threat_identified_by_abnormal_security_command,
+            'abnormal-security-manage-abnormal-case':
+                manage_an_abnormal_case_command,
+            'abnormal-security-submit-inquiry-to-request-a-report-on-misjudgement-by-abnormal-security':
```
Comment (Shellyber) [inline Packs/AbnormalSecurity/Integrations/AbnormalSecurity/AbnormalSecurity.py:313]:
```suggestion
            'abnormal-security-submit-inquiry-to-request-a-report-on-misjudgement':
```

**Review by Shellyber — COMMENTED** [review]
Looks Good! 👍 
Please see my last comments.

**Review by Shellyber — APPROVED** [review]

**Pull request merged**